libraries/AP_InertialSensor/AP_InertialSensor_Invensense.h

```diff
@@ -122,6 +122,24 @@ class AP_InertialSensor_Invensense : public AP_InertialSensor_Backend
     // are we doing more than 1kHz sampling?
     bool _fast_sampling;
 
+    // use a sample counter to "timestamp" samples from the sensor
+    // Since the system timebase is not synchronized to the sensor sample clock,
+    // this is the best way to identify missing samples in the log. With this
+    // counter supplied to the notify methods, the SampleUS field becomes an integer
+    // sequence which increments each time a sample is read from the FIFO. As long as
+    // the FIFO is read at a sufficient rate, there should be no dropped samples
+    // between the sensor and the call to notify, with the exception of FIFO corruption:
+    // if corruption is detected, _seqcnt is zeroed to indicate a break in continuity.
+    uint32_t _seqcnt;
+
+#define INVENSENSE_DEBUG 1
+//#undef INVENSENSE_DEBUG
+#if defined(INVENSENSE_DEBUG)
+    unsigned _dcnt = 0;
```

Comment:
In general, we always avoid using "unsigned" by itself. Since we work on many hardware platforms with different compilers/linkers it's important for the source to be concise so it works equally on the different platforms.